component/otelcol/receiver/jaeger/jaeger.go

```diff
@@ -0,0 +1,211 @@
+// Package jaeger provides an otelcol.receiver.jaeger component.
+package jaeger
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/alecthomas/units"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/component/otelcol/receiver"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jaegerreceiver"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelconfig "go.opentelemetry.io/collector/config"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name: "otelcol.receiver.jaeger",
+		Args: Arguments{},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := jaegerreceiver.NewFactory()
+			return receiver.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.receiver.jaeger component.
+type Arguments struct {
+	Protocols      ProtocolsArguments       `river:"protocols,block"`
+	RemoteSampling *RemoteSamplingArguments `river:"remote_sampling,block,optional"`
+
+	// Output configures where to send received data. Required.
+	Output *otelcol.ConsumerArguments `river:"output,block"`
+}
+
+var (
+	_ river.Unmarshaler  = (*Arguments)(nil)
+	_ receiver.Arguments = Arguments{}
+)
+
+// DefaultArguments provides default settings for Arguments. All protocols are
+// configured with defaults and then set to nil in UnmarshalRiver if they were
+// not defined in the source config.
+var DefaultArguments = Arguments{
+	Protocols: ProtocolsArguments{
+		GRPC: &otelcol.GRPCServerArguments{
+			Endpoint:  "0.0.0.0:14250",
+			Transport: "tcp",
+		},
+		ThriftHTTP: &otelcol.HTTPServerArguments{
+			Endpoint: "0.0.0.0:14268",
+		},
+		ThriftBinary: &ProtocolUDP{
+			Endpoint:      "0.0.0.0:6831",
+			QueueSize:     1_000,
+			MaxPacketSize: 65 * units.KiB,
+			Workers:       10,
+		},
+		ThriftCompact: &ProtocolUDP{
+			Endpoint:      "0.0.0.0:6832",
```

Comment:
These two are swapped - `ThriftBinary` should default to port 6832, and `ThriftCompact` 6831.

(see https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/jaegerreceiver for where they're documented in the OTelCol docs)